presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -88,6 +88,15 @@ public static long length(@SqlType("varchar(x)") Slice slice)
         return countCodePoints(slice);
     }
 
+    @Description("count of code points of the given string")
+    @ScalarFunction("length")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.BIGINT)
+    public static long charLength(@LiteralParameter("x") Long x, @SqlType("char(x)") Slice slice)
```

Comment:
This argument is resolved by ImplementationDependency mechanism. This interface forces to return Object. Additionally whole dependency mechanism use `Long` on multiple levels. Without changes in this system it is not possible to pass primitive `long`.